### DISCUSSION TITLE
[vm] Deprecate language module?

### DISCUSSION THREAD

**Discussion by oowekyala** [pmd/pmd#3121] [General]
I'm not sure whether the VM language module is worth the effort of keeping it up to date with PMD 7.x. Nobody mentioned it on github/sourceforge/stackoverflow since 2014 ---I think [this is the most recent mention of it](https://stackoverflow.com/questions/27275994/maven-pmd-plugin-3-3-with-vm-and-xml-rule-sets), and there's nothing about it on either sourceforge or github apart from us keeping it up to date.

Besides, the codebase looks copy-pasted from the original velocity repo, with many unused things that only made sense in that codebase. It duplicates Javacc-generated code, duplicates functionality of AbstractNode, etc. I think it will get worse and worse unless we rehaul it somehow. But without a community showing interest in it, is this really worthwhile? I would argue for deprecating it unless someone takes it over

**Comment by adangel** [comment]
Are there any users for the VM (Apache Velocity) language out there and uses the [built-in rules](https://pmd.github.io/latest/pmd_rules_vm.html) or custom rules? Please let us know.
